src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/FormattingLanguageServerClient.cs

```diff
@@ -105,6 +115,48 @@ private RazorDocumentRangeFormattingResponse Format(RazorDocumentRangeFormatting
                 var generatedHtml = codeDocument.GetHtmlDocument().GeneratedHtml;
                 generatedHtml = generatedHtml.Replace("\r", "", StringComparison.Ordinal).Replace("\n", "\r\n", StringComparison.Ordinal);
 
+                var editHandlerAssembly = Assembly.Load("Microsoft.WebTools.Languages.LanguageServer.Server, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
+                var editHandlerType = editHandlerAssembly.GetType("Microsoft.WebTools.Languages.LanguageServer.Server.Html.OperationHandlers.ApplyFormatEditsHandler", throwOnError: true);
+                var bufferManagerType = editHandlerAssembly.GetType("Microsoft.WebTools.Languages.LanguageServer.Server.Shared.Buffer.BufferManager", throwOnError: true);
+
+                var contentTypeService = new FakeContentTypeRegistryService();
+                contentTypeService.AddContentType(HtmlContentTypeDefinition.HtmlContentType, new[] { StandardContentTypeNames.Text });
+
+                var textBufferFactoryService = new FakeTextBufferFactoryService(contentTypeService);
+                var textBufferListeners = Array.Empty<Lazy<IWebTextBufferListener, IOrderedComponentContentTypes>>();
+                var bufferManager = Activator.CreateInstance(bufferManagerType, new object[] { contentTypeService, textBufferFactoryService, textBufferListeners });
+                var threadSwitcher = (IThreadSwitcher)Activator.CreateInstance(
+                    Assembly.Load("Microsoft.WebTools.Shared, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a").GetType("Microsoft.WebTools.Shared.Threading.JoinableTaskFactoryThreadSwitcher", throwOnError: true),
+                    new object[] { new JoinableTaskContext().Factory });
+                var applyFormatEditsHandler = Activator.CreateInstance(editHandlerType, new object[] { bufferManager, threadSwitcher, textBufferFactoryService });
+
+                // Make sure the buffer manager knows about the source document
+                var documentUri = DocumentUri.From($"file:///{@params.HostDocumentFilePath}");
+                var contentTypeName = HtmlContentTypeDefinition.HtmlContentType;
+                var initialContent = generatedHtml;
+                var snapshotVersionFromLSP = 0;
+                Assert.IsAssignableFrom<ITextSnapshot>(bufferManager.GetType().GetMethod("CreateBuffer").Invoke(bufferManager, new object[] { documentUri, contentTypeName, initialContent, snapshotVersionFromLSP }));
+
+                var requestType = editHandlerAssembly.GetType("Microsoft.WebTools.Languages.LanguageServer.Server.ContainedLanguage.ApplyFormatEditsParamForOmniSharp", throwOnError: true);
+                var serializedValue = $@"{{
```

Comment:
If you don't want to write this out manually you can serialize the [actual type](https://github.com/OmniSharp/csharp-language-server-protocol/blob/323e325c1e900e0d97a5681816bc9d34c3fd89a0/src/Protocol/Models/FormattingOptions.cs)